akd/src/append_only_zks.rs

```diff
@@ -326,10 +328,32 @@ impl Azks {
         let (left_insertion_set, right_insertion_set) =
             partition_longest_common_prefix(insertion_set, current_node.label);
 
+        // if parallel: spawn a tokio task for the left child
+        #[cfg(feature = "parallel_insert")]
+        let maybe_future = if !left_insertion_set.is_empty() {
+            let storage_clone = storage.clone();
+            let left_child_label = current_node.get_child_label(Direction::Left)?;
+            Some(tokio::task::spawn(async move {
```

Comment:
Can we try vlocking_spawn and limit the concurrency level to say a few levels of the tree? say like 3 levels? That would limit to 7 dedicated threads each with a decent amount of work